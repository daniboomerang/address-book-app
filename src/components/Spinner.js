import React from 'react';
import cx from 'classnames';
import styles from './Spinner.module.css';

/**
 * It renders an animated loading spinner.
 */
const Spinner = () => (
  <div className={cx(styles.spinner, 'border-t-4 border-red-600 rounded-full w-10 h-10')} />
);

export default Spinner;
