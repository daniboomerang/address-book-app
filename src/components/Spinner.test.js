import React from 'react';
import { mount } from 'enzyme';
import Spinner from './Spinner';

describe('Spinner component', () => {
  it('should render correctly', () => {
    const wrapper = mount(<Spinner />);

    expect(wrapper.find('.spinner').length).toBe(1);
    expect(wrapper).toMatchSnapshot();
  });
});
