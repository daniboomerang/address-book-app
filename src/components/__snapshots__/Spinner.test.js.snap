// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component should render correctly 1`] = `
<Spinner>
  <div
    className="spinner border-t-4 border-red-600 rounded-full w-10 h-10"
  />
</Spinner>
`;
